app/scrambler.py

```diff
@@ -1,17 +1,69 @@
+from random import choice, shuffle
+
 from picture import Picture
+from tile import Tile
 
 
 class Scrambler:
     """Scramble a picture in different ways."""
 
-    def scramble_rows(self, picture: Picture) -> None:
+    @classmethod
+    def scramble_rows(cls, picture: Picture) -> None:
         """Split an Image up into rows an rearranges them."""
-        raise NotImplementedError
+        num_tiles = 8
+        image_size = picture.image.size
+        tile_size = (image_size[0], int(image_size[1] / num_tiles))
 
-    def scramble_grid(self, picture: Picture) -> None:
-        """Split an Image up into tiles an rearranges them."""
-        raise NotImplementedError
+        for y in range(num_tiles):
+            x0, y0, x1, y1 = (0, tile_size[1] * y, tile_size[0], tile_size[1] * (y + 1))
+            picture.tiles[y] = Tile(picture.image.crop((x0, y0, x1, y1)), (x0, y0))
+
+        picture.tile_order = list(picture.tiles.keys())
+        shuffle(picture.tile_order)
+
+    @classmethod
+    def scramble_grid(cls, picture: Picture) -> None:
+        """Split an Image up into tiles an rearranges them and rotates each tile randomly."""
```

Comment:
If you have to use and twice, might be too many things going on in this method and should be split.

Reply:
Rotation and rearranges are happening in a same loop, so I think its better to keep them in same method. Adding multiple methods will increase complexity for this scenario.